packages/backend/src/open_payments/invoice/service.ts

```diff
@@ -94,6 +98,25 @@ async function createInvoice(
   }
 }
 
+async function handleInvoicePayment(
+  deps: ServiceDependencies,
+  invoiceId: string
+): Promise<void> {
+  const amountReceived = await deps.accountingService.getTotalReceived(
+    invoiceId
+  )
+  if (!amountReceived) {
+    return
+  }
+  const invoice = await Invoice.query(deps.knex).findById(invoiceId)
+  if (invoice.amount <= amountReceived) {
+    await invoice.$query(deps.knex).patch({
+      active: false
+    })
```

Comment:
Would it cause problems if this query + patch are in separate transactions?

Reply:
I don't think so, but this could be put in a single
```
    await Invoice.query(deps.knex).patch({
      active: false
    }).where('id', invoiceId).andWhere('amount', '<=', amountReceived)
```

Reply:
I ran into objection/knex type errors trying to compare with a bigint in `andWhere` so I've put the query and patch in a transaction https://github.com/interledger/rafiki/pull/197/commits/88ba6d1f2c0cb97ca9cdbaaac747fa8b548798dd

Reply:
Did you try passing the `toString` of the bigint?

Reply:
Nope :upside_down_face: 
https://github.com/interledger/rafiki/pull/197/commits/6478831124e22700d05251d02dc131a7b3b3618b